cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -325,29 +337,17 @@ class SnappyUnifiedMemoryManager private[memory](
 
   override def dropStorageMemoryForObject(name: String,
                                           memoryMode: MemoryMode,
-                                          ignoreNumBytes: Long): Long =
-    synchronized {
```

Comment:
synchronized not required? And the executionPool, storagePool determination not required? Didn't understand this removal of ON_HEAP/OFF_HEAP check.